First version of the Station Login-event

LudvigStridsberg opened the pull request:
Added the file for the event and "required" it in the event-index-file.

LudvigStridsberg committed:
First version of the Station Login-event

Added the file for the event and "required" it in the event-index-file.

trumtomte commented on events/station-login.js:
```diff
@@ -0,0 +1,55 @@
+    const game = context.getGameState();
+
+    // Get the station- and playerobject so that we can get their attributes
+    const stationObj = game.stations[station];
```
Check that the station exists (just in case), ie. `stationObj !== undefined`.

trumtomte commented on events/station-login.js:
```diff
@@ -0,0 +1,55 @@
+
+    // Get the station- and playerobject so that we can get their attributes
+    const stationObj = game.stations[station];
+    const player = game.players[context.id()];
```
Since you use the playerId att more then one place, make a variable out of.

trumtomte commented on events/station-login.js:
```diff
@@ -0,0 +1,55 @@
+      }
+    // We're not in the play phase
+    if (game.properties.phase.type !== 2) {
+    context.send('event:station:login:fail', { errorCode: 1 });
```
Did you run `npx eslint . --quiet`? Seems like the indentation is a bit off here.

trumtomte commented on events/station-login.js:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    // Calculate the total login-time
+    const loginTime = ((stationObj.properties.loginTime * stationObj.properties.loginMultiplier) * player.properties.loginTimeMultiplier);
```
Perfect!

trumtomte commented on events/station-login.js:
```diff
@@ -0,0 +1,55 @@
+
+    // Create a timestamp for the start of the login
+    const date = new Date();
+    const start = date.getTime();
```
Pro tip: You can get the time with even less code: `const start = Date.now();`.

trumtomte commented on events/station-login.js:
```diff
@@ -0,0 +1,55 @@
+
+    // After the defined time the user is allowed into the station
+    setTimeout(function() {
+        const game = context.getGameState();
```
Yes! It's great that you re-fetch the game like this, great thinking 👍🏻

trumtomte reviewed (changes requested):
Some small changes needs to be fixed.

LudvigStridsberg committed:
Second version of the Station Login-event

Implemented the requested changes.
Added a variable for the player's id and changed the code so that it uses it, fixed some indentation-mistakes, added another error-check and shortened the part of the code that gets a timestamp.

Co-Authored-By: Sebastian Bengtegård <sebastianbengtegard@gmail.com>

trumtomte reviewed (approved):
💯

Commit:
Merge branch 'main' into Event_Station_login
